import_events.py

```diff
@@ -0,0 +1,235 @@
+from os import path
+from datetime import datetime
+import json
+import boto.s3
+import boto.provider
+import postgres
+
+with open("config.json") as f:
+    CONFIG = json.loads(f.read())
+
+if "aws_access_key_id" not in CONFIG:
+    aws = boto.provider.Provider("aws")
+    CONFIG["aws_access_key_id"] = aws.get_access_key()
+    CONFIG["aws_secret_access_key"] = aws.get_secret_key()
+
+S3_BUCKET = "net-mozaws-prod-us-west-2-pipeline-analysis"
+DB_URI = "postgresql://{db_username}:{db_password}@{db_host}:{db_port}/{db_name}".format(**CONFIG)
+
+# The default data set automatically expires data at
+# three months. We also have sampled data sets that
+# cover a longer history.
+SAMPLE_RATES = (
+    {"percent":10, "months":24, "suffix":"_sampled_10"},
+    {"percent":50, "months":6, "suffix":"_sampled_50"},
+    {"percent":100, "months":3, "suffix":""}
+)
+
+# The temporary table receives raw data from S3.
+# The permenant table then receives data appropriately typed.
+TABLE_NAMES = {
+    "temp":"temporary_raw_{event_type}_data",
+    "perm":"{event_type}_events{suffix}"
+}
+
+Q_DROP_TEMPORARY_TABLE = """
+    DROP TABLE IF EXISTS {table};
+""".format(table=TABLE_NAMES["temp"])
+
+Q_CREATE_EVENTS_TABLE = """
+    CREATE TABLE IF NOT EXISTS {table} (
+        timestamp TIMESTAMP NOT NULL SORTKEY ENCODE lzo,
+        {schema}
+    );
+""".format(table=TABLE_NAMES["perm"],
+           schema="{schema}")
```

Comment:
This partial string-formatting quickly gets out of hand you'll notice. I imagine there must be a cleaner way to do it, feel free to suggest something. Maybe I've gone too far in trying to refactor the table names out of the queries. My aim was to only have to write each table name once, so that if they change they only have to change in one place too.